### PR TITLE
G Suite: Remove statuses shown for Google Workspace mailboxes pending ToS acceptance

### PR DESCRIPTION
**Pull request by stephanethomas**
This pull request, together with D73732-code, removes the statuses shown for Google Workspace mailboxes pending ToS acceptance. As a reminder, a user who purchased a Google Workspace account has to agree to two different types of terms of service: the ones for the account and the ones for the mailbox itself. We were using this last bit of information to prompt users to log into their account. Unfortunately, this information is unreliable and we had several cases where it would be wrong. Since we cannot trust it, this pull request makes the following changes:

* It no longer displays the `Action required` status on the `Domains` page when a Google account has mailboxes pending ToS acceptance. This was confusing because in this context this status should be referring to the domain itself, and not any email solution that depends on it. Moreover, this status would also be displayed at the top of the `Domain Settings` page without additional information, meaning the user had no idea how to actually fix that state.
* It updates the `Email Plan` page so it no longer show mailboxes with a `Pending` status (and its red border) when the account is pending ToS acceptance. I wasn't planning to make that change originally but keeping this status seemed much harder to achieve (because of how our logic is implemented) rather than just removing it. We still display a clear warning and CTA in the header of this page so this shouldn't be a problem though.
* It updates the `Email Plan` page to no longer show an `Action required` status (and its red border) as well as a `Finish setup` button for mailboxes pending ToS acceptance. As mentioned above, this information is unreliable and could be confusing.
* Finally, this pull request removes the components that would display a modal box, and offer users to reset their password. This modal box was no longer accessible anyway, so it didn't make sense to keep that code around. If we want to offer the option to reset a password again, we will rather add it to the `Email Plan` page.

##### `Domains` page

Before | After
------ | -----
DOMAINS - BEFORE | DOMAINS - AFTER

##### `Email Plan` page for account pending ToS acceptance

Before | After
------ | -----
EMAIL PLAN GS - BEFORE | EMAIL PLAN GS - AFTER

##### `Email Plan` page for mailbox pending ToS acceptance

Before | After
------ | -----
EMAIL PLAN GW - BEFORE| EMAIL PLAN GW - AFTER

#### Testing instructions

> Note you should apply D73732-code to your sandbox if it has not been merged yet (and point `public-api.wordpress.com` to your sandbox in that case).

1. Run `git checkout update/google-workspace-warnings` and start your server, or access a [live branch](https://github.com/Automattic/wp-calypso/pull/60460#issuecomment-1021395461)
2. Open the [`Emails` page](http://calypso.localhost:3000/email)
3. Purchase a Google Workspace account, and wait until it is provisioned
4. Navigate to the [`Domains` page](http://calypso.localhost:3000/domains/manage)
5. Assert that the status is `Active` for the domain with this account
6. Head back to the `Emails` page
7. Assert that the status is `Action required` for your new account
8. Click that account to access the `Email Plan` page
9. Assert that there is warning with a `Finish setup` button in the header
10. Assert that there is no button nor status in the list of mailboxes
11. Click the `Finish setup` button
12. Log into your Google Admin Console, and accept both ToS
13. Refresh the `Email Plan` page
14. Assert that the warning in the header has disappeared

### CODE REVIEW

File: client/lib/emails/email-provider-constants.js [hunk 1]
```diff
@@ -6,7 +6,5 @@ export const EMAIL_ACCOUNT_TYPE_TITAN_MAIL_EXTERNAL = 'titan_external';
 export const EMAIL_TYPE_FORWARD = 'email_forward';
 export const EMAIL_USER_ROLE_ADMIN = 'admin';
 export const EMAIL_WARNING_SLUG_GOOGLE_ACCOUNT_TOS = 'google_pending_tos_acceptance';
-export const EMAIL_WARNING_SLUG_GOOGLE_MAILBOX_TOS = 'google_user_not_agreed_to_terms';
 export const EMAIL_WARNING_SLUG_UNUSED_MAILBOXES = 'unused_mailboxes';
 export const EMAIL_WARNING_SLUG_UNVERIFIED_FORWARDS = 'unverified_forwards';
-export const EMAIL_WARNING_TYPE_ACTION_REQUIRED = 'action_required';
```
Comment (stephanethomas) [inline client/lib/emails/email-provider-constants.js:12]: This particular constant was never used.

**Comment by AllTerrainDeveloper** [comment]
I am not sure but the test is not fully working for me...

### EOT

**Domains**

Before
image

After
image

**Emails**
Before
image

After
image

**Review by AllTerrainDeveloper — CHANGES_REQUESTED** [review]
@stephanethomas seems that I have failing test. We can discuss it on Monday :)

**Comment by leonardost** [comment]
I noticed yesterday that I had some domains with "Action required" status but no highlighted actions shown in the domain settings page, so thank you @stephanethomas for this fix 🙂 looks good from the domains' side, but I got the same results as Daniel in the emails screen:

| Before | After |
| ------ | ------ |
| Screen Shot 2022-01-28 at 19 17 38 | Screen Shot 2022-01-28 at 19 17 43 |
| Screen Shot 2022-01-28 at 19 20 19 | Screen Shot 2022-01-28 at 19 20 25 |

**Comment by stephanethomas** [comment]
I think you're getting a different behavior because you didn't apply the server-side patch D73732-code. That's my fault though, I've only mentioned that patch in the description, and forgot about the testing instructions. I've updated them now.

**Comment by AllTerrainDeveloper** [comment]
@stephanethomas this is not working as expected, I'll approve but you need to rebase first :) 

### EOT

image

**Review by AllTerrainDeveloper — APPROVED** [review]

**Pull request merged**